### PR TITLE
Added Support for Spherical ST_Area

### PR DESCRIPTION
**Pull request by kokosing**
Added Support for Spherical ST_Area

Fixed OH and PA as they had duplicate vertices

STArea Benchmark :

Result "com.facebook.presto.plugin.geospatial.BenchmarkSTArea.stSphericalArea500k":
  328652966.567 ±(99.9%) 63645108.583 ns/op [Average]
  (min, avg, max) = (237252788.000, 328652966.567, 516285755.000), stdev = 73293801.693
  CI (99.9%): [265007857.984, 392298075.149] (assumes normal distribution)

Benchmark                            Mode  Cnt          Score          Error  Units
BenchmarkSTArea.stArea               avgt   20     137845.475 ±    28021.370  ns/op
BenchmarkSTArea.stArea500k           avgt   20   16663361.799 ±  2491691.163  ns/op
BenchmarkSTArea.stSphericalArea      avgt   20    3250484.666 ±   840187.416  ns/op
BenchmarkSTArea.stSphericalArea500k  avgt   20  328652966.567 ± 63645108.583  ns/op

Extracted from: https://github.com/prestodb/presto/pull/12315

### CODE REVIEW

File: presto-geospatial/src/test/resources/us-states.tsv [hunk 1]
```diff
@@ -32,10 +32,10 @@ NM	POLYGON ((-107.421329 37.000263, -106.868158 36.994786, -104.337812 36.994786
...
-OH	POLYGON ((-80.518598 41.978802, -80.518598 40.636951, -80.666475 40.582182, -80.595275 40.472643, -80.600752 40.319289, -80.737675 40.078303, -80.830783 39.711348, -81.219646 39.388209, -81.345616 39.344393, -81.455155 39.410117, -81.57017 39.267716, -81.685186 39.273193, -81.811156 39.0815, -81.783771 38.966484, -81.887833 38.873376, -82.03571 39.026731, -82.221926 38.785745, -82.172634 38.632391, -82.293127 38.577622, -82.331465 38.446175, -82.594358 38.424267, -82.731282 38.561191, -82.846298 38.588575, -82.890113 38.758361, -83.032514 38.725499, -83.142052 38.626914, -83.519961 38.703591, -83.678792 38.632391, -83.903347 38.769315, -84.215533 38.807653, -84.231963 38.895284, -84.43461 39.103408, -84.817996 39.103408, -84.801565 40.500028, -84.807042 41.694001, -83.454238 41.732339, -83.065375 41.595416, -82.933929 41.513262, -82.835344 41.589939, -82.616266 41.431108, -82.479343 41.381815, -82.013803 41.513262, -81.739956 41.485877, -81.444201 41.672093, -81.011523 41.852832, -80.518598 41.978802, -80.518598 41.978802))
```
Comment (martint) [inline presto-geospatial/src/test/resources/us-states.tsv]: These seem like unrelated changes.

File: presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java [hunk 2]
```diff
@@ -1491,6 +1494,166 @@ private static void validateSphericalType(String function, OGCGeometry geometry,
...
+        if (end - start < 3) {
+            // A path with less than 3 distinct points is not valid for calculating an area
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Polygon is not valid: a loop contains less then 3 vertices.");
+        }
+
+        Point point = new Point();
+
+        // Initialize the calculator with the last point
+        polygon.getPoint(end - 1, point);
+        SphericalExcessCalculator calculator = new SphericalExcessCalculator(point);
```
Comment (martint) [inline presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java]: A class for this seems unnecessary. It could all be done in a method (maybe just here?) that takes the list of points or the polygon and start/end points. It would also simplify some oddities, like keeping track of whether the computation is done.

File: presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        data.setup();
+        BenchmarkSTArea benchmark = new BenchmarkSTArea();
+
+        assertEquals(Math.round(1000 * (Double) benchmark.stSphericalArea(data) / 3.659E8), 1000);
+        assertEquals(Math.round(1000 * (Double) benchmark.stSphericalArea500k(data) / 38842273735.0), 1000);
+        assertEquals(benchmark.stArea(data), 0.05033099592771004);
+        assertEquals(Math.round(1000 * (Double) benchmark.stArea500k(data) / Math.PI), 1000);
+    }
+
+    private static String createPolygon(double numVertices)
```
Comment (martint) [inline presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java]: numVertices -> vertexCount

File: presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    {
+        StringBuilder sb = new StringBuilder();
```
Comment (martint) [inline presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java]: sb -> builder

File: presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java (hunk 3 above)
Comment (martint) [reply presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java]: Also, the argument should be of type `int`

File: presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        sb.append("POLYGON((");
+        String separator = "";
+        for (int i = 0; i < numVertices; i++) {
+            double angle = i * Math.PI * 2 / numVertices;
+            sb.append(separator);
+            sb.append(Math.cos(angle));
+            sb.append(" ");
+            sb.append(Math.sin(angle));
+            separator = ",";
```
Comment (martint) [inline presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/BenchmarkSTArea.java]: Instead of setting separator to `","` after the first iteration, it'd be clearer to append it conditionally if `i > 0`. 

Alternatively, this could be written as:

```java
IntStream.range(0, numVertices)
        .mapToObj(i -> {
            double angle = i * Math.PI * 2 / numVertices;
            return Math.cos(angle) + " " + Math.sin(angle);
        })
        .collect(Collectors.joining(","));
```

File: presto-geospatial/src/test/resources/us-states.tsv (hunk 1 above)
Comment (kokosing) [reply presto-geospatial/src/test/resources/us-states.tsv]: It is needed, otherwise it fails `throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Polygon is not valid: it has two identical consecutive vertices");`

**Comment by kokosing** [comment]
@martint Comments addressed.

**Comment by kokosing** [comment]
@martint Is there anything else you would me to fix here?
